relentless/ensemble.py

```diff
@@ -1,176 +1,437 @@
 
-    def reset(self):
-        if 'V' in self.conjugates:
+        """
+        return self._constant
+
+    def clear(self):
+        """Clears all conjugate variables and all rdf values in the ensemble
+        (set to `None`).
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            Returns the ensemble object with cleared parameters.
+
+        """
+        if not self.constant['V']:
             self._V = None
-        if 'P' in self.conjugates:
+        if not self.constant['P']:
             self._P = None
         for t in self.types:
-            if 'N_{}'.format(t) in self.conjugates:
+            if not self.constant['N'][t]:
                 self._N[t] = None
-            if 'mu_{}'.format(t) in self.conjugates:
+            if not self.constant['mu'][t]:
                 self._mu[t] = None
-        self.rdf = core.PairMatrix(self.types)
+        for pair in self.rdf:
+            self.rdf[pair] = None
 
         return self
 
     def copy(self):
-        ens = copy.deepcopy(self)
-        return ens.reset()
-
-    def save(self, basename='ensemble'):
-        # dump thermo data to json file
-        data = {'types': self.types,
-                'kB': self.kB,
-                'T': self.T,
-                'P': self.P,
-                'V': self.V,
-                'N': self.N,
-                'mu': self.mu,
-                'conjugates': self.conjugates}
-        with open('{}.json'.format(basename),'w') as f:
-            json.dump(data, f, sort_keys=True, indent=4)
-
-        # dump rdfs in separate files
-        for pair in self.rdf:
-            if self.rdf[pair] is not None:
-                i,j = pair
-                np.savetxt('{}.{}.{}.dat'.format(basename,i,j), self.rdf[pair].table, header='r g[{},{}](r)'.format(i,j))
+        """Makes a copy of the ensemble.
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            A copy of the ensemble object.
+
+        """
+        return copy.deepcopy(self)
+
+    def save(self, filename):
+        """Saves the ensemble parameters and the RDF values into a JSON file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = []
+
+        # append thermo data
+        thermo = {'T': self.T,
+                  'P': self.P,
+                  'V': self.V,
+                  'mu':self.mu,
+                  'N': self.N,
+                  'kB': self.kB,
+                  'constant': self.constant} #need constant?
+        data.append(thermo)
+
+        # append rdf data
+        rdf = {pair:self.rdf[pair].table for pair in self.rdf}
+        data.append(rdf)
+
+        # dump data to json file
+        with open('{}.json'.format(filename),'w') as f:
+            json.dump(data, f, indent=4)
 
     @classmethod
-    def load(self, basename='ensemble'):
-        with open('{}.json'.format(basename)) as f:
+    def load(self, filename):
```

Comment:
Let's rename this `from_file` so that it's clear that it is actually making a new class.